lib/jekyll/site_data/directory.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module SiteData
+    class Directory
+      # Delegate given (zero-arity) method(s) to the Hash object stored in instance
+      # variable `@meta`.
+      def self.delegate_to_meta(*symbols)
+        symbols.each { |sym| define_method(sym) { @meta.send(sym) } }
+      end
+      private_class_method :delegate_to_meta
+
+      delegate_to_meta :freeze, :inspect
+      attr_accessor :context
+
+      def initialize
+        @meta = {}
```

Comment:
`@meta` is a very generic name which doesn't explain what this variable is used for. Could we use a more descriptive name that tells us what it holds?

Reply:
The `Hash` instance assigned to the variable contains the *data* of constituent data files. I had therefore considered naming the variable `@data`, then `@metadata` and finally just `@meta`.
*Naming things is hard* :cry:
I will try to come up with better names..

Reply:
It's `@files` right? in DataReader, the code does `data[key] = DataFile.new...` which stores into this hash by method_missing. https://github.com/jekyll/jekyll/pull/8771/files#diff-ae0ebeb5a3a44f07ee61a0e5851d165c646ddab160824029c467bffea3717e7bR51

I'd call it `@files` or something like that since that's what it's storing? Is it storing anything else?

Reply:
@parkr, `@files` makes me think of an Array of file objects like how we have an array of staticfiles. But `DataFile` isn't technically a file object either, only a container for the "parsed data" of the actual data file on disk. The closest I think that will express the intention correctly is calling this variable a `@registry` and `DataFile` to be `DataEntry` instead.